### PR TITLE
feat: pm auth connector integration - Plaid

### PR DESCRIPTION
**Pull request by Sanskar2001**
## Type of Change

- [ ] Bugfix
- [x] New feature
- [ ] Enhancement
- [ ] Refactoring
- [ ] Dependency updates
- [ ] Documentation
- [ ] CI/CD

## Description

Added pm auth connector - added support for multiple pm auth connector - 

Currently supported - Plaid

## How did you test it?

Adding a pm auth connector and opening the plaid SDK.

## Checklist

- [x] I ran `npm run re:build`
- [x] I reviewed submitted code
- [ ] I added unit tests for my changes where possible

### CODE REVIEW

**Review by seekshiva — DISMISSED** [review]

**Review by PritishBudhiraja — DISMISSED** [review]

File: src/Payments/VerifyBankDetails.res [hunk 1]
```diff
@@ -0,0 +1,104 @@
...
+    )
+  })
+
+  let submitCallback = React.useCallback((ev: Window.event) => {
+    let json = ev.data->JSON.parseExn
+    let confirm = json->getDictFromJson->ConfirmType.itemToObjMapper
+    if confirm.doSubmit {
+      postFailedSubmitResponse(
+        ~errortype="validation_error",
+        ~message="Payment cannot be performed. Please go to saved PM Screen to update your payment method.",
```
Comment (vsrivatsa-juspay) [inline src/Payments/VerifyBankDetails.res:70]:
```suggestion
        ~message="Please add bank details first.",
```

File: src/Payments/VerifyBankDetails.res [hunk 2]
```diff
@@ -0,0 +1,104 @@
...
+      padding: "20px",
+      cursor: "pointer",
+      borderRadius: themeObj.borderRadius,
+      borderColor: themeObj.borderColor,
+      borderWidth: "2px",
+    }>
+    {if showLoader {
+      <Loader />
+    } else {
+      {React.string("Verify Bank Details")}
```
Comment (vsrivatsa-juspay) [inline src/Payments/VerifyBankDetails.res:101]:
```suggestion
      {React.string("Add Bank Details")}
```

File: src/Utilities/PaymentHelpers.res [hunk 3]
```diff
@@ -1717,3 +1713,134 @@ let paymentIntentForPaymentSession = (
...
+  ~clientSecret,
+  ~paymentMethodType,
+  ~pmAuthConnectorsArr,
+  ~iframeId,
+) => {
+  open Promise
+  let endpoint = ApiEndpoint.getApiEndPoint()
+  let uri = `${endpoint}/payment_methods/auth/link`
+  let headers = [("Content-Type", "application/json"), ("api-key", publishableKey)]->Dict.fromArray
+
```
Comment (vsrivatsa-juspay) [inline src/Utilities/PaymentHelpers.res:1728]: add logs for API request, response and errors

Comment (PritishBudhiraja) [reply src/Utilities/PaymentHelpers.res:1728]: `add logs for API request, response and errors`
Will do in the 2nd PR.

**Review by seekshiva — APPROVED** [review]

**Review by vsrivatsa-juspay — APPROVED** [review]

**Pull request merged**